Sources/Hexiconjuror/Commands/Output Strings/OutputStrings.swift

```diff
@@ -0,0 +1,221 @@
+    private let parser = StringsParser()
+
+    func run(_ options: OutputStrings.Options) -> Result<(), ConjurorError> {
+        let sourcePath = options.definitionsPath.isEmpty ? environment.projectPath : environment.projectPath.appendingPathComponent(options.definitionsPath)
+        let sourceFiles = FileManager.default.enumerator(at: sourcePath, includingPropertiesForKeys: [.isRegularFileKey])?
+            .compactMap { $0 as? URL }
+            .filter { $0.pathExtension == "swift" } ?? []
+        let resourcesPath = options.resourcesPath.isEmpty ? environment.projectPath : environment.projectPath.appendingPathComponent(options.resourcesPath)
+        let stringsFiles = FileManager.default.enumerator(at: resourcesPath, includingPropertiesForKeys: [.isRegularFileKey])?
+            .compactMap { $0 as? URL }
+            .filter { $0.pathExtension == "strings" } ?? []
+        let temporaryDirectory = FileManager.default.temporaryDirectory.appendingPathComponent(UUID().uuidString)
+        return Result {
+            try FileManager.default.createDirectory(atPath: temporaryDirectory.path, withIntermediateDirectories: false, attributes: nil)
+        }
+            .flatMapCatching { files in
+                let localize = Process()
+                let arguments = "/usr/bin/xcrun extractLocStrings -o \(temporaryDirectory.path)"
+                    .split(separator: " ").map(String.init)
+                    + sourceFiles.map { $0.path }
+                localize.executableURL = URL(fileURLWithPath: arguments[0])
+                localize.arguments = Array(arguments.dropFirst())
+                try localize.run()
+                localize.waitUntilExit()
+            }
+            .flatMapCatching {
+                let existingTables = try stringsFiles.map {
+                    try Table(url: $0, strings: parser.parse(file: $0))
+                }
+                let newTables = try FileManager.default.contentsOfDirectory(at: temporaryDirectory, includingPropertiesForKeys: [.isRegularFileKey], options: [])
+                    .filter { $0.pathExtension == "strings" }
+                    .map { return try Table(url: $0, strings: parser.parse(file: $0)) }
+                return Array(newTables.map { newTable -> [Table] in
+                    let matches = existingTables.filter { $0.name == newTable.name }
+                    guard !matches.isEmpty else {
+                        return [mutate(newTable) { $0.url = resourcesPath.appendingPathComponent("\($0.name).strings") }]
+                    }
+                    return matches.map {
+                        var matchingTable = $0
+                        newTable.strings.forEach { newString in
+                            guard var string = matchingTable.strings[newString.key] else {
+                                matchingTable.strings[newString.key] = newString.value
```

Comment:
not required in any way but you can return the setting of a value which evaluates to void to make this else one line